packages/edit-site/src/components/sidebar-navigation-screen/index.js

```diff
@@ -51,7 +56,12 @@ export default function SidebarNavigationScreen( {
 	return (
 		<>
 			<VStack
-				className="edit-site-sidebar-navigation-screen__main"
+				className={ classnames(
+					'edit-site-sidebar-navigation-screen__main',
+					{
+						'has-footer': !! footer,
```

Comment:
Seems a bit brute force, but it's all I could think of to ensure the gap is consistent. 

The footer needs a top margin to allow the top grey border some room to display clearly.

Reply:
While it is a little brute force, I like the clarity of `has-footer` as a classname, and it's clear what's going on here. Looks good to me for now!